archive/tar_test.go

```diff
@@ -35,24 +36,40 @@ func testTar(t *testing.T, when spec.G, it spec.S) {
 		h.AssertNil(t, os.RemoveAll(tmpDir))
 	})
 
-	when("#ReadTarArchive", func() {
+	when("#UntarLayer", func() {
 		var pathModes = []archive.PathMode{
 			{"root", os.ModeDir + 0755},
 			{"root/readonly", os.ModeDir + 0500},
 			{"root/standarddir", os.ModeDir + 0755},
 			{"root/standarddir/somefile", 0644},
 			{"root/readonly/readonlysub/somefile", 0444},
 			{"root/readonly/readonlysub", os.ModeDir + 0500},
+			{"Files", os.ModeDir + 0755},
+			{"Files/standardfile_Files", 0644},
+			{"Hives", os.ModeDir + 0755},
+			{"Hives/standardfile_Hives", 0644},
+		}
+
+		// Golang for Windows only implements owner permissions
+		if runtime.GOOS == "windows" {
+			pathModes = []archive.PathMode{
+				{"root", os.ModeDir + 0777},
+				{"root/readonly", os.ModeDir + 0555},
+				{"root/standarddir", os.ModeDir + 0777},
+				{"root/standarddir/somefile", 0666},
+				{"root/readonly/readonlysub/somefile", 0444},
+				{"root/readonly/readonlysub", os.ModeDir + 0555},
```

Comment:
Should tar entries with names that don't start with `Files/` get written to disk in the windows case?

Reply:
Also, I am surprised this works with forward slashes on windows. Does `os.Stat` convert these for us on windows?

Reply:
Yeah good catch about the slashes. I'll look into this... It's probably just not trying to chmod.